libevmasm/RuleList.h

```diff
@@ -174,6 +174,18 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleList(
 		});
 	}
 
+	for (auto const& op: std::vector<Instruction>{
+		Instruction::ADDRESS,
+		Instruction::CALLER,
+		Instruction::ORIGIN,
+		Instruction::COINBASE
+	})
+		rules.push_back({
+			{Instruction::AND, {{u256("0xffffffffffffffffffffffffffffffffffffffff"), op}}},
```

Comment:
Do we need to add both `and(0xfff, x)` and `and(x, 0xfff`)?

Reply:
Yeah, we should, forgot its not doing that anymore on its own.